goagen/gen_app/generator.go

```diff
@@ -415,3 +417,81 @@ func (g *Generator) generateUserTypes() error {
 	}
 	return utWr.FormatCode()
 }
+
+// mediaTypeMetadataImports will grab all imports specified in a struct:field:type Metadata tag and
+// add them to the imports for the media_types.go file.
+func (g *Generator) mediaTypeMetadataImports(s []*codegen.ImportSpec) []*codegen.ImportSpec {
```

Comment:
I'm not sure what you are getting at here. Besides the fact that we should probably have recursion.